salt/modules/consul.py

```diff
@@ -78,9 +78,12 @@ def _query(function,
     base_url = urllib.parse.urljoin(consul_url, '{0}/'.format(api_version))
     url = urllib.parse.urljoin(base_url, function, False)
 
-    if data is None:
-        data = {}
-    data = salt.utils.json.dumps(data)
+    if method == 'GET':
```

Comment:
Is there a chance that the `GET` method would ever be used with `data`?

Reply:
There's nothing to say that a body payload cannot be sent in a `GET` request.
Your mileage may vary.
IMHO it's not a particularly smart thing to do.  Tornado is dogmatic about it and will complain.

```
>>> data=json.dumps({})
>>> data
'{}'
>>> res=salt.utils.http.query('https://github.com/saltstack/salt/blob/develop/salt/utils/http.py',method='GET',data=data)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python2.7/site-packages/salt/utils/http.py", line 515, in query
    **req_kwargs
  File "/usr/lib64/python2.7/site-packages/tornado/httpclient.py", line 102, in fetch
    self._async_client.fetch, request, **kwargs))
  File "/usr/lib64/python2.7/site-packages/tornado/ioloop.py", line 444, in run_sync
    return future_cell[0].result()
  File "/usr/lib64/python2.7/site-packages/tornado/concurrent.py", line 214, in result
    raise_exc_info(self._exc_info)
  File "<string>", line 3, in raise_exc_info
ValueError: Body must be None for GET request
>>>
```
The following code solves the bug - and also permits a payload to be sent with `GET`:

```
    if method != 'GET' or data is not None:
        if data is None:
            data = {}
        data = salt.utils.json.dumps(data)
```
I'll commit that if you prefer.

Reply:
I just want to make sure we don't have a regression/change in expected behavior here for someone. The default for `method` is `GET` in the function signature, so it seems like we would have run into this error before. At the very least the docs should probably be updated.

@garethgreenaway It looks like you've been involved in this module in the past. I'd like to have you look this over as well.

Reply:
@campbellmc Can you update the docs about this change as I mentioned above? After that is done, we can get this merged in.

Reply:
Hi there!
Sorry for the delay.  I added a comment in docstring for _query.  I hope that's what you meant for me to do.
Campbell